src/Exception/BatchException.php

```diff
@@ -2,61 +2,136 @@
 
 namespace Http\Client\Exception;
 
+use Http\Client\BatchResult;
+use Http\Client\Exception;
+use Psr\Http\Message\RequestInterface;
+
 /**
  * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
  */
-final class BatchException extends TransferException
+final class BatchException extends RuntimeException
```

Comment:
lets document this: This exception is thrown when a batch of requests led to at least one failure. It holds the response/exception pairs and gives access to a BatchResponse with the successful responses.